packages/ast-tooling/index.ts

```diff
@@ -48,34 +48,72 @@ export type {
 	ChildNode as HtmlChildNode,
 
 	// js
-	AstTypes,
-	AstKinds,
+	TsEstree as AstTypes,
 
 	//css
 	CssChildNode
 };
 
-export function parseScript(content: string): AstTypes.Program {
-	const recastOutput: { program: AstTypes.Program } = recastParse(content, {
-		parser: {
-			parse: tsParse
+/**
+ * Parses as string to an AST. Code below is taken from `esrap` to ensure compatibilty.
+ */
+export function parseScript(content: string): TsEstree.Program {
+	const comments: TsEstree.Comment[] = [];
+
+	const acornTs = acorn.Parser.extend(tsPlugin({ allowSatisfies: true }));
+
+	const ast = acornTs.parse(content, {
+		ecmaVersion: 'latest',
+		sourceType: 'module',
+		locations: true,
+		onComment: (block, value, start, end) => {
+			if (block && /\n/.test(value)) {
+				let a = start;
+				while (a > 0 && content[a - 1] !== '\n') a -= 1;
+
+				let b = a;
+				while (/[ \t]/.test(content[b])) b += 1;
+
+				const indentation = content.slice(a, b);
+				value = value.replace(new RegExp(`^${indentation}`, 'gm'), '');
+			}
+
+			comments.push({ type: block ? 'Block' : 'Line', value, start, end });
+		}
+	}) as TsEstree.Program;
+
+	Walker.walk(ast as TsEstree.Node, null, {
+		_(commentNode, { next }) {
+			let comment = comments.shift();
+
+			while (comment && comment.start! < commentNode.start!) {
+				commentNode.leadingComments ??= [];
+				commentNode.leadingComments.push(comment);
+				comment = comments.shift();
+			}
+
+			next();
+
+			comment = comments.shift();
+			if (comment) {
+				const slice = content.slice(commentNode.end, comment.start);
+
+				if (/^[,) \t]*$/.test(slice)) {
+					commentNode.trailingComments = [comment];
+				}
+			}
 		}
 	});
```

Comment:
Adding a short description of what's going on here would be super helpful as well.

Reply:
I have absolutely no clue tbh. 

The code is taken from here: https://github.com/sveltejs/esrap/blob/9daf5dd43b31f17f596aa7da91678f2650666dd0/test/common.js#L12 and there is also simmilar code inside svelte itself in this file https://github.com/sveltejs/svelte/blob/701f085c82d11e6064433731d36b33d4894c706a/packages/svelte/src/compiler/phases/1-parse/acorn.js#L6

We could potentially borrow one of the comments from the svelte source, as it's at least a bit more clear with that one. We could also try to closer align to the svelte source code if that's preferable.

Reply:
Improved comments a bit